app/src/main/java/com/example/asiantech/playmusic/service/MusicService.java

```diff
@@ -0,0 +1,323 @@
+                    pausePlayer();
+                    stopForeground(true);
+                } else {
+                    playNext();
+                }
+                break;
+            case 2:
+                playSong();
+                break;
+            case 3:
+                playRandom();
+                break;
+            default:
+                playNext();
+        }
+    }
+
+    @Override
+    public boolean onError(MediaPlayer mp, int what, int extra) {
+        Log.d("TAG SERVICE", "onError");
+        return false;
+    }
+
+    public void playSong() {
+        Log.d("TAG SERVICE", "playSong");
+        mMessage = "reset";
+        sendBroadcast();
+        mMediaPlayer.reset();
+        Uri uri = ContentUris.withAppendedId(MediaStore.Audio.Media.EXTERNAL_CONTENT_URI,
+                mPlayList.get(mSongPosition).getId());
+        try {
+            mMediaPlayer.setDataSource(getApplicationContext(), uri);
+        } catch (IOException e) {
+            Log.e("MUSIC SERVICE", "Error set data source");
+        }
+        mMediaPlayer.prepareAsync();
+    }
+
+    @Override
+    public void onPrepared(MediaPlayer mp) {
+        Log.d("TAG SERVICE", "onPrepared");
+        mp.start();
+        updateNotification();
+        mMessage = "play";
+
+        sendBroadcast();
+    }
+
+    public void setPlayList(ArrayList<Song> listSong) {
+        Log.d("TAG SERVICE", "setPlayList");
+        mPlayList = listSong;
+    }
+
+    public ArrayList<Song> getPlayList() {
+        return mPlayList;
+    }
+
+    public void setListType(String type) {
+        mListType = type;
+    }
+
+    public String getListType() {
+        return mListType;
+    }
+
+    public void setSong(int position) {
+        Log.d("TAG SERVICE", "setSong");
+        this.mSongPosition = position;
+    }
+
+    public long getSongPlayingId() {
+        return mPlayList.get(mSongPosition).getId();
+    }
+
+    public int getSongPosition() {
+        return mSongPosition;
+    }
+
+    private void createNotification() {
+        Log.d("TAG SERVICE", "createNotification");
+        mRemoteViews = new RemoteViews(getPackageName(), R.layout.notification_play_music);
+        NotificationCompat.Builder builder = new NotificationCompat.Builder(this)
+                .setSmallIcon(R.drawable.icon_notification)
+                .setContentTitle("play music")
+                .setContent(mRemoteViews);
+        Intent intent = new Intent(this, MainActivity_.class);
+        intent.putExtra("message", "fromNotify");
+        PendingIntent pendingIntent = PendingIntent.getActivity(this, 0, intent,
+                PendingIntent.FLAG_UPDATE_CURRENT);
+        builder.setContentIntent(pendingIntent);
+        setListenerNotification();
+        mNotification = builder.build();
+    }
+
+    private void setListenerNotification() {
+        Intent intent = new Intent(this, MusicReceiver.class);
+        intent.putExtra("action", "close");
```

Comment:
WARNING: [AvoidDuplicateLiterals] The String literal "action" appears 4 times in this file; the first occurrence is on line 206
http://pmd.sourceforge.net/pmd-5.2.3/pmd-java/rules/java/strings.html#AvoidDuplicateLiterals